homeassistant/components/freebox/switch.py

```diff
@@ -0,0 +1,63 @@
+"""Support for Freebox Delta, Revolution and Mini 4K."""
+import logging
+
+from homeassistant.components.freebox import DATA_FREEBOX
+from homeassistant.components.switch import SwitchDevice
+
+DEPENDENCIES = ['freebox']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up the switch."""
+    fbx = hass.data[DATA_FREEBOX]
+    async_add_entities([FbxWifiSwitch(fbx)], True)
+
+
+class FbxWifiSwitch(SwitchDevice):
+    """Representation of a freebox wifi switch."""
+
+    def __init__(self, fbx):
+        """Initilize the Wifi switch."""
+        self._name = 'Freebox WiFi'
+        self._state = None
+        self._fbx = fbx
+
+    @property
+    def name(self):
+        """Return the name of the switch."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
+
+    async def _async_set_state(self, enabled):
+        """Turn the switch on or off."""
+        from aiofreepybox.exceptions import InsufficientPermissionsError
+
+        wifi_config = {"enabled": enabled}
+        try:
+            await self._fbx.wifi.set_global_config(wifi_config)
+        except InsufficientPermissionsError:
+            _LOGGER.warning('Home Assistant does not have permissions to'
+                            ' modify the Freebox settings. Please refer'
+                            ' to documentation. https://home-assistant.'
+                            '.io/components/switch.freebox/')
```

Comment:
I think it is better to avoid the URL since those are hard to keep current and I think this one might even be obsolete already(?)

Reply:
Yep, will change this.